webhookbot/webhookbot/http.go

```diff
@@ -83,7 +80,39 @@ func (h *HTTPSrv) handleHook(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	h.Stats.Count("handle - success")
-	h.ChatEcho(hook.convID, "[hook: *%s*]\n\n%s", hook.name, msg)
+	if _, err := h.Config().KBC.SendMessageByConvID(hook.convID, "[hook: *%s*]\n\n%s", hook.name, msg); err != nil {
+		// error created in https://github.com/keybase/client/blob/1985b18c4e7659bede1d4a2e68e4f68467acebc6/go/client/chat_svc_handler.go#L1407
+		// error created in https://github.com/keybase/keybase/blob/9a82c96231ea2c6132532002e58bac80849265e6/go/chatbase/storage/sql_chat.go#L2324
+		if strings.Contains(err.Error(), "no conversations matched") ||
+			strings.Contains(err.Error(), "GetConvTriple called with unknown ConversationID") {
+			h.Debug("ChatEcho: failed to send echo message: %s", err)
+			return
+		}
+
+		// error created in https://github.com/keybase/client/blob/7d6aa64f3fba66adba7a5dd1cc7c523d5086a548/go/chat/msgchecker/plaintext_checker.go#L50
+		toBig, regexErr := regexp.MatchString("message of size [0-9]+ bytes exceeds the maximum length of [0-9]+ bytes",
+			err.Error())
+		if regexErr != nil {
+			h.Errorf("regex matching error: %s", regexErr)
+		}
+
+		if toBig {
+			fileName := fmt.Sprintf("webhookbot-%s-%d.txt", hook.name, time.Now().Unix())
+			filePath := fmt.Sprintf("/tmp/%s", fileName)
+			defer os.Remove(filePath)
```

Comment:
I don't think this will work as is, I don't think that `SendAttachment` is fully blocking, it could still be trying to read the file when we remove it. I had a problem with this in `handlePProf` and delayed the file removal

Reply:
I didn't see that! Thanks, I'll do the same here.